aten/src/ATen/native/cuda/Normalization.cuh

```diff
@@ -765,22 +765,22 @@ std::tuple<Tensor, Tensor, Tensor, Tensor> batch_norm_backward_reduce_cuda_templ
     mean_dy_ = at::empty_like(mean_);
     mean_dy_xmu_ = at::empty_like(mean_);
   }
-  auto grad_options = grad_out_.options();
+  auto grad_options = mean_.options();
```

Comment:
Not sure if we discussed it here before.
IIRC, `mean_` is passed here as calculated from the batch_norm_gather_stats_with_counts https://github.com/pytorch/pytorch/blob/b93f29ded33baee8f337a5b3dd30a142de726da8/torch/nn/modules/_functions.py#L62-L70

If we have input and weight both in fp16. This will break?
We might have to pass in an optional variable (either weight or bias) here to make the decision. It cannot be deduced from given information.

Reply:
But I could imagine that we must have fp16 layer with fp16 input. If unit test is fine, I can take another look at this with a clearer head tomorrow morning :)

Reply:
This makes sense and I'm passing now the `weight` tensor to this method. Since it can be `None`, I'm calling `weight_.options()` now for `grad_weight_` and `grad_bias_` separately.

Reply:
Looks good. Thanks for the hard work.